### ISSUE TITLE
Cloudflare Notification Policy - magic_tunnel_health_check_event not working.

### ISSUE THREAD

**Issue by miteshchauhan** [cloudflare/terraform-provider-cloudflare#3272]
### Confirmation

- [X] This is a bug with an existing resource and is not a feature request or enhancement. Feature requests should be submitted with Cloudflare Support or your account team.
- [X] I have searched the issue tracker and my issue isn't already found.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

Terraform v1.7.5
on darwin_arm64
+ provider registry.terraform.io/cloudflare/cloudflare v4.30.0

### Affected resource(s)

cloudflare_notification_policy
alert_type = "magic_tunnel_health_check_event"

### Terraform configuration files

```hcl
resource "cloudflare_notification_policy" "magic-tunnel-health" {
  account_id = local.stepstone_cloudflare_account_id
  name       = "TF - Magic Tunnel Health Check Alert"
  alert_type = "magic_tunnel_health_check_event"
  enabled    = true
  email_integration {
    id = "yyy"
  }
  email_integration {
    id = "xxx"
  }
  filters {
    tunnel_id = ["zzz"]
    slo = ["90"]
  }
}
```

### Link to debug output

N/A

### Expected output

resource should be applied successfully

### Actual output

│ Error: error creating policy TF - Magic Tunnel Health Check Alert: Invalid object selections (17003)
│ 
│   with cloudflare_notification_policy.magic-tunnel-health,
│   on magic-tunnel-health.tf line 1, in resource "cloudflare_notification_policy" "magic-tunnel-health":
│    1: resource "cloudflare_notification_policy" "magic-tunnel-health" {
│ 

### Steps to reproduce

1. try to deploy any magic-tunnel health check will give you the error. it's not clear what the requirements are for this type of check (was hoping to have it managed within TF then we could update the config based on changes made on the Dash portal)
2. try to import any magic-tunnel health check will give a different error

### Additional factoids

We also face issues when importing pre-existing magic-tunnel-health checks into an empty resource. We get the following error message:

 Error: failed to set filters: Invalid address to set: []string{"filters", "0", "tunnel_name"}

Ticket also raised within CF No#3221552

**Comment by Nmishin** [comment]
I see that `magic_tunnel_health_check_event` was added in this PR https://github.com/cloudflare/terraform-provider-cloudflare/pull/2937 by @jasoncabot 
and released as a part of 4.20.0 version

Unfortunately I don't have that type of alerts in my CF account, and I can't test/fix it.
@jasoncabot  Jason, maybe you can?

**Comment by jeffsani** [comment]
Posting my debug log for the same issue.  We are missing something in the provider that is required by the Tunnel Health Check API - [](https://developers.cloudflare.com/api/operations/notification-policies-create-a-notification-policy)

Error: Unsupported argument
on cf_aws_mwan_tunnels_and_routes.tf line 72, in resource "cloudflare_notification_policy" "AWS-VPN-1-Notification":
    tunnel_name       = ["AWS-VPN-1"]
An argument named **"tunnel_name"** is not expected here.


[run-y3QFHPNd7eTSgjmt-plan-log.txt](https://github.com/user-attachments/files/15873159/run-y3QFHPNd7eTSgjmt-plan-log.txt)
